api/src/main/java/org/cardanofoundation/rosetta/api/network/service/impl/NetworkServiceImpl.java

```diff
@@ -199,4 +199,43 @@ private List<Producer> getPublicRoots(List<PublicRoot> publicRoots) {
 
   }
 
+  @Override
+  public void verifyNetworkRequest(final NetworkIdentifier networkIdentifier) {
+    if (networkIdentifier != null) {
+      if (!verifyBlockchain(networkIdentifier.getBlockchain())) {
+        throw ExceptionFactory.invalidBlockchainError();
+      }
+      if (!verifyNetwork(networkIdentifier.getNetwork())) {
+        throw ExceptionFactory.networkNotFoundError();
+      }
+    }
+  }
+
+  private boolean verifyBlockchain(String blockchain) {
+    return blockchain.equals(Constants.CARDANO_BLOCKCHAIN);
+  }
+
+  private boolean verifyNetwork(String network) {
+    Network supportedNetwork = getSupportedNetwork();
+
+    switch ((int) supportedNetwork.getProtocolMagic()) {
+      case Constants.MAINNET_PROTOCOL_MAGIC -> {
+        return network.equalsIgnoreCase(Constants.MAINNET);
+      }
+      case Constants.TESTNET_PROTOCOL_MAGIC -> {
+        return network.equalsIgnoreCase(Constants.TESTNET);
+      }
+      case Constants.PREPROD_PROTOCOL_MAGIC -> {
+        return network.equals(Constants.PREPROD);
+      }
+      case Constants.PREVIEW_PROTOCOL_MAGIC -> {
+        return network.equals(Constants.PREVIEW);
+      }
+      case Constants.DEVKIT_PROTOCOL_MAGIC -> {
+        return network.equals(Constants.DEVKIT);
+      }
+      default -> throw ExceptionFactory.networkNotFoundError();
```

Comment:
Considering returning a boolean, it's not clear that here could an exception be thrown. 
Let's either throw exceptions within verify methods or just in the initial one. 
*the first option will be clearer since the method `verifyNetworkRequest` returns void.